### DISCUSSION TITLE
My repository detected language is incorrect. How to edit?

### DISCUSSION THREAD

**Discussion by ZedZipDev** [community/community#35313] [New to GitHub] [answered]
My repository is
https://github.com/ZedZipDev/SQLCLR-SpellChecker
I see on the right side the detected languages are 

Smalltalk 77.6% -- ?
 C# 18.9%
 TSQL 3.5%

There is no any Smalltalk code. How to correct languages?

**Comment by ihnatenkoo** [comment]
Hello. ZedZipDev. See here:
[how to change the category?](https://github.com/orgs/community/discussions/23333rl)

**Reply by ZedZipDev** [reply]
Ok, but the difference with their approach is: 
They have these files and exclude them, ok
*.css linguist-detectable=false
*.js linguist-detectable=true
but I have no any Smalltalk files in the project.  So what should I exclude? 

PS. Слава Україні!

**Reply by ihnatenkoo** [reply]
Возможно стоить попробовать исключить все файлы с детекта, и добавить только то расширение которое хочешь чтоб было отмечено в репозитории, может быть так поможет:

>  \* linguist-detectable=false
>  *.html linguist-detectable=true

PS. Слава Україні!

**Reply by ZedZipDev** [reply]
У меня только файлы *.cs и несколько скриптов *.sql.
Откуда -то лезет Smalltalk. 
т.е. типа такого надо сделать:

* linguist-detectable=false
*.cs linguist-detectable=true
*.sql linguist-detectable=true

Попробую

**Reply by ZedZipDev** [reply]
Сделал так, получил
SQL 99.6%
Other 0.4%
Шарп не видит.

Ну, хоть так ;-)

**Reply by ihnatenkoo** [reply]
P.S Вперед до перемоги!

**Reply by ZedZipDev** [reply]
ок, отмечу, но вопрос то в общем остался

**Reply by ZedZipDev** [reply]
Again after the .gitignore adding:

Smalltalk 77.6% <-????
C# 18.9%
TSQL 3.5%

Why Smalltalk? I do not use it.

**Reply by ihnatenkoo** [reply]
проверь первую строчку в файле .gitattributes
там нужно попробовать указать что все файлы переводишь в detectable=false

**Reply by ZedZipDev** [reply]
Вот содержимое

linguist-detectable=false
*.cs linguist-detectable=true
*.sql linguist-detectable=true

А это ссылка на файл

https://github.com/ZedZipDev/SQLCLR-SpellChecker/blob/main/.gitattributes

Может ты увидишь какую-то ошибку.

**Reply by ihnatenkoo** [reply]
первая строка, попробуй так, нужна звездочка чтобы указать что эта команда должна действовать на все файлы
 \* linguist-detectable=false

**Reply by ZedZipDev** [reply]
Сделал: 
SQL   99.6%
Other 0.4%

Какой-то нереальний п-ц :-)

**Comment by ZedZipDev** [comment, marked as answer]
The solution is 
*.cs linguist-language=C#
